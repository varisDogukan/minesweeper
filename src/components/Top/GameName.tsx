import styled from "@emotion/styled";

export interface GameNameProps {
  children: string;
}

export const GamaName = styled.h1`
  font-size: 2em;
`;
