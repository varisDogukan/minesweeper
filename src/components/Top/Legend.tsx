import React from "react";
import styled from "@emotion/styled";

export interface LegendProps {
  feature: string;
  firstAction: string;
}

export default function Legend({ feature, firstAction }: LegendProps) {
  return (
    <Parent>
      <strong>{feature}: </strong>
      <FlagComboParent>
        <FirstAction>{firstAction}</FirstAction>
      </FlagComboParent>
    </Parent>
  );
}

const FlagComboParent = styled.code`
  background-color: #e3e3e3;
`;

const Parent = styled.legend`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`;

const FirstAction = styled.span`
  color: #ec433c;
`;
