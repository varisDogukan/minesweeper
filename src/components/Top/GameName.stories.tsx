import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import GameName, { GameNameProps } from "./GameName";

const meta: Meta<typeof GameName> = {
  title: "Top/GameName",
  component: GameName,
};

export default meta;
type Story = StoryObj<GameNameProps>;

export const MinesweeperGameName: Story = {
  render: (args: any) => <GameName {...args} />,
  args: {
    name: "Minesweeper",
  },
};
