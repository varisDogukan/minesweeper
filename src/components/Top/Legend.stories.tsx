import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import Legend, { LegendProps } from "./Legend";

const meta: Meta<typeof Legend> = {
  title: "Top/Legend",
  component: Legend,
};

export default meta;
type Story = StoryObj<LegendProps>;

export const GameLegend: Story = {
  render: (args: any) => <Legend {...args} />,
  args: {
    feature: "Flag",
    firstAction: "ctrl",
    secondAction: "click",
  },
};
