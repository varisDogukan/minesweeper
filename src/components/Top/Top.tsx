import React from "react";
import styled from "@emotion/styled";

import Legend, { LegendProps } from "./Legend";
import { GamaName, GameNameProps } from "./GameName";

export type TopComponentType = LegendProps & GameNameProps;

export default function Top({ children, ...legendProps }: TopComponentType) {
  return (
    <Header>
      <GamaName>{children}</GamaName>
      <Legend {...legendProps} />
    </Header>
  );
}

const Header = styled.header`
  text-align: center;
  position: relative;
  display: inline-block;
`;
