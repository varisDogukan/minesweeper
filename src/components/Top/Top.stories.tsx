import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import Top, { TopComponentType } from "./Top";

const meta: Meta<typeof Top> = {
  title: "Top/Top",
  component: Top,
};

export default meta;
type Story = StoryObj<TopComponentType>;

export const GameLegend: Story = {
  render: (args: any) => <Top {...args} />,
  args: {
    children: "Minesweeper",
    feature: "Flag",
    firstAction: "Ctrl",
  },
};
