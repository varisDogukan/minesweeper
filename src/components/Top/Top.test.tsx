import React from "react";
import { render } from "@testing-library/react";

import Top from "./Top";

it("GameName renders correctyle", () => {
  const { asFragment } = render(
    <Top
      children='Minesweeper'
      feature='Flag'
      firstAction='Ctrl'
      secondAction='Click'
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
