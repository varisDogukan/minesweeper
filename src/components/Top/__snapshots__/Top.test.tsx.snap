// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameName renders correctyle 1`] = `
<DocumentFragment>
  <header
    class="css-dn2347"
  >
    <h1
      class="css-18kvaxm"
    >
      Minesweeper
    </h1>
    <legend
      class="css-17qz6a2"
    >
      <strong>
        Flag: 
      </strong>
      <code
        class="css-h9xasx"
      >
        <span
          class="css-udfepp"
        >
          Ctrl
        </span>
         + 
        <span
          class="css-9c23k1"
        >
          Click
        </span>
      </code>
    </legend>
  </header>
</DocumentFragment>
`;
